yellowbrick/features/base.py

```diff
@@ -307,3 +428,57 @@ def _determine_target_color_type(self, y):
                 "could not determine target color type "
                 "from target='{}' to '{}'"
             ).format(self.target_type, self._target_color_type))
+
+    def get_target_color_type(self):
+        """
+        Returns the computed target color type if fitted or specified by the user.
+        """
+        if self._target_color_type is None:
+            raise NotFitted("unknown target color type on unfitted visualizer")
+        return self._target_color_type
+
+    def get_colors(self, y):
```

Comment:
This is the primary new functionality - because the DataVisualizer determines the target color type I thought it was also important for it to be able to compute the colors based on the target. This will also reduce a lot of duplicated functionality in the subclasses.